### PR TITLE
Transform abstract classes added

### PR DESCRIPTION
**Pull request by MatInGit**
This pull request introduces a new abstract base class for transformers and refactors an existing transformer to inherit from this base class.

### CODE REVIEW

**Comment by MatInGit** [comment]
Also removed a copy of `cli.py` that I accidentaly created in the repo root.

**Comment by Wikyboy** [comment]
Looking at the code in [BaseTransformers](https://github.com/MatInGit/lume-deployment/blob/e7492b0eb35fec5535ecd535d063ff947545feaa/model_manager/src/transformers/BaseTransformers.py#L9C1-L21C1)

The following doc string can be confusing. I would recommend type hinting. If it is a dictionary in a dictionary I would recommanend splitting it into two variables, a dictionary or list of formulas and a symbols list.


```python
class SimpleTransformer(BaseTransformer):
    def __init__(self, config):
        """
        config: dict
            dictionary containing the following keys:
            - variables: dict
                dictionary containing the following keys:
                - formula: str
                    formula to be used for transformation
            - symbols: list
                list of symbols to be used in the formula
        """
``` 

```python
class SimpleTransformer(BaseTransformer):
    def __init__(self, config: dict[dict[str,str],str]):
```

File: model_manager/src/transformers/BaseTransformer.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+
+class BaseTransformer:
+    @abstractmethod
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py:4]: Type hinting for the config as well just to clarify that the object passed is a dictionary and not string to the .yml dictionary.

File: model_manager/src/transformers/BaseTransformer.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        pass
+    
+    @abstractmethod
+    def transform(self):
+        """
+        Initiate transform function to transform the input data
+        """
+        pass
+    
+    @abstractmethod
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py:18]: Type hinting and description of what the objects are:
* pv_name: is it a list of PVs? if so, what kind of object, is it string?
* value: what is meant by values here?

File: model_manager/src/transformers/BaseTransformer.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+
+class BaseTransformer:
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py:3]: Type hinting for the config as well just to clarify that the object passed is a dictionary and not string to the .yml dictionary.

File: model_manager/src/transformers/BaseTransformer.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        config: dict passed from the pv_mappings.yaml files
+        """
+        pass
+    
+    @abstractmethod
+    def transform(self):
+        """
+        Initiate transform function to transform the input data
+        """
+        pass
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py:16]: Type hinting and description of what the objects are:
* pv_name: is it a list of PVs? if so, what kind of object, is it string?
* value: what is meant by values here?

File: model_manager/src/transformers/BaseTransformer.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from abc import ABC, abstractmethod
+
+class BaseTransformer:
+    @abstractmethod
+    def __init__(self, config):
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py]: Type hinting for the config as well just to clarify that the object passed is a dictionary and not string to the .yml dictionary.

File: model_manager/src/transformers/BaseTransformer.py [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    
+    @abstractmethod
+    def transform(self):
+        """
+        Initiate transform function to transform the input data
+        """
+        pass
+    
+    @abstractmethod
+    def handler(self, pv_name, value):
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py]: Type hinting and description of what the objects are:
* pv_name: is it a list of PVs? if so, what kind of object, is it string?
* value: what is meant by values here?

File: model_manager/src/transformers/BaseTransformer.py [hunk 7]
```diff
@@ -2,21 +2,21 @@
...
     @abstractmethod
     def transform(self):
         """
-        Initiate transform function to transform the input data
+        Call transform function to transform the input data, see SimpleTransformer in model_manager/src/transformers/BaseTransformers.py for an example.
         """
         pass
     
     @abstractmethod
-    def handler(self, pv_name, value):
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py]: Add description of what the objects are:
* pv_name and the value associated with that PV
* this comes from me being ignorant but currently it cant handle arrays?

File: model_manager/src/transformers/BaseTransformer.py [hunk 8]
```diff
@@ -2,21 +2,21 @@
...
+    def handler(self, pv_name: str, value: dict | float | int):
```
Comment (Wikyboy) [inline model_manager/src/transformers/BaseTransformer.py:19]: Add description of what the objects are in docstring:
* pv_name and the value associated with that PV
* this comes from me being ignorant but currently it cant handle arrays?

The way this method is implemented [SimpleTransformer](https://github.com/MatInGit/lume-deployment/blob/e7492b0eb35fec5535ecd535d063ff947545feaa/model_manager/src/transformers/BaseTransformers.py#L45C1-L66C1) class is confusing. They seem to be dictionaries and not primitive types.

Comment (MatInGit) [reply model_manager/src/transformers/BaseTransformer.py:20]: I think we are ok to merge, its been a while...

**Pull request merged**